lib/Sema/ConstraintSystem.h

```diff
@@ -1452,9 +1452,10 @@ class ConstraintSystem {
   /// Get the type for an expression.
   Type getType(const Expr *E) const {
     assert(hasType(E) && "Expected type to have been set!");
-    assert((!E->getType() ||
-            E->getType()->isEqual(ExprTypes.find(E)->second)) &&
-           "Mismatched types!");
+    // FIXME: lvalue differences
```

Comment:
This particular one will go away (hopefully soon!). There are some other asserts that need to be reenabled and investigated and/or modified if the differences are legit and expected, or the underlying problems need to be fixed.